### PR TITLE
add flag to disable jetstream job

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

Add flag to disable jetstream job 

## Related Issues

https://github.com/astronomer/issues/issues/6245

## Testing

QA should pass to disable jetstream hook job
```
nats:
  nats:
      createJetStreamJob: false 
```

## Merging

cherry-pick to release-0.34,release-0.35

### CODE REVIEW

**Review by rishkarajgi — APPROVED** [review]

File: charts/nats/templates/jetstream-job.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ########################################################
 ## Astronomer Platform Namespace label configure Hook ##
 ########################################################
-{{- if or  .Values.nats.jetstream.enabled .Values.global.nats.jetStream.enabled  .Values.global.stan.enabled }}
+{{- if and (or  .Values.nats.jetstream.enabled .Values.global.nats.jetStream.enabled  .Values.global.stan.enabled) ( .Values.nats.createJetStreamJob ) }}
```
Comment (danielhoherd) [inline charts/nats/templates/jetstream-job.yaml:4]:
```suggestion
{{- if and (or .Values.nats.jetstream.enabled .Values.global.nats.jetStream.enabled .Values.global.stan.enabled) ( .Values.nats.createJetStreamJob ) }}
```

File: charts/nats/values.yaml [hunk 2]
```diff
@@ -19,6 +19,9 @@ images:
 
 
 nats:
+
+  createJetStreamJob: true
```
Comment (danielhoherd) [inline charts/nats/values.yaml:23]:
```suggestion
  createJetStreamJob: true
```

File: tests/chart_tests/test_nats_jetstream.py [hunk 3]
```diff
@@ -145,3 +145,18 @@ def test_stan_with_jetstream_disabled(self, kube_version):
         )
 
         assert len(docs) == 4
+
+    def test_jetstream_hook_job_disabled(self, kube_version):
```
Comment (danielhoherd) [inline tests/chart_tests/test_nats_jetstream.py:149]: We need tests with `"global": {"nats": {"jetStream": {"enabled": False}}}` that test the behavior of `createJetStreamJob`

Comment (pgvishnuram) [reply tests/chart_tests/test_nats_jetstream.py:149]: createJetStreamJob - the test case added validates these

Comment (pgvishnuram) [reply tests/chart_tests/test_nats_jetstream.py:149]: we are testing them as a part of TestNatsJetstream

Comment (danielhoherd) [reply tests/chart_tests/test_nats_jetstream.py:149]: As long as we have coverage of all 4 of those possibilities then we're good.

File: charts/nats/templates/jetstream-job.yaml [hunk 4]
```diff
@@ -1,7 +1,8 @@
 ########################################################
 ## Astronomer Platform Namespace label configure Hook ##
 ########################################################
-{{- if or  .Values.nats.jetstream.enabled .Values.global.nats.jetStream.enabled  .Values.global.stan.enabled }}
+{{- if  .Values.nats.createJetStreamJob  -}}
+{{- if or  .Values.nats.jetstream.enabled .Values.global.nats.jetStream.enabled  .Values.global.stan.enabled  }}
```
Comment (danielhoherd) [inline charts/nats/templates/jetstream-job.yaml:5]:
```suggestion
{{- if or .Values.nats.jetstream.enabled .Values.global.nats.jetStream.enabled .Values.global.stan.enabled }}
```

Comment (danielhoherd) [inline charts/nats/templates/jetstream-job.yaml:5]:
```suggestion
{{- if .Values.nats.createJetStreamJob -}}
{{- if or .Values.nats.jetstream.enabled .Values.global.nats.jetStream.enabled .Values.global.stan.enabled }}
```

**Pull request merged**